graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasJanusElement.java

```diff
@@ -105,23 +110,45 @@ public Set<String> getPropertyKeys() {
 
     @Override
     public void removeProperty(String propertyName) {
+        RequestContext context = RequestContext.get();
         Iterator<? extends Property<String>> it = getWrappedElement().properties(propertyName);
         while(it.hasNext()) {
             Property<String> property = it.next();
             property.remove();
+            //fill the map for the entityGuid and remove value
+            if (propertyName.startsWith(INTERNAL_PROPERTY_KEY_PREFIX)) {
+                String entityGuid = this.getProperty(GUID_PROPERTY_KEY, String.class);
+                if (StringUtils.isNotEmpty(entityGuid)) {
+                    context.getAllInternalAttributesMap().get(entityGuid).put(propertyName, null);
+                }
+            }
         }
     }
 
     @Override
     public void removePropertyValue(String propertyName, Object propertyValue) {
+        RequestContext context = RequestContext.get();
         Iterator<? extends Property<Object>> it = getWrappedElement().properties(propertyName);
-
+        // we create a map of properties that are only internal
         while (it.hasNext()) {
             Property currentProperty      = it.next();
             Object   currentPropertyValue = currentProperty.value();
 
             if (Objects.equals(currentPropertyValue, propertyValue)) {
                 currentProperty.remove();
+                //fill the map for the entityGuid and remove value
+                if (propertyName.startsWith(INTERNAL_PROPERTY_KEY_PREFIX)) {
+                    String entityGuid = this.getProperty(GUID_PROPERTY_KEY, String.class);
+                    if (StringUtils.isNotEmpty(entityGuid)) {
+                        if (context.getAllInternalAttributesMap().get(entityGuid) != null) {
+                            context.getAllInternalAttributesMap().get(entityGuid).put(propertyName, null);
```

Comment:
This can not be null.
The value must be always `ALL_ELEMENTS - propertyValue`.

Imagine `__meaningNames` current has values `["term_0", "term_1"]`,
Someone updates asset to remove `term_0`,
final value for `__meaningNames` must be `["term_0", "term_1"]` - `term_0` = `["term_1"]`, not null

Reply:
I think can alter the while loop to avoid `break` & iterate all elements to collect them in a list by skipping when `currentProperty.remove();` is hit.
After all items are iterated, we can put the prepared list in `context.getAllInternalAttributesMap().get(entityGuid)`

This way we will put list on final state of the value for property